### PR TITLE
The Grand Reformat

### PR DESCRIPTION
**Pull request by mattgodbolt**
- everything made prettier...literally
- some tweaks to include a few more files, including documentation
- minor changes to format style
- some tiny `// prettier-ignore` changes to keep a few things the way we like them
- a couple of super minor tweaks to embedded document types to ensure they format correctly

### CODE REVIEW

File: static/panes/diff.js [hunk 1]
```diff
@@ -126,94 +120,104 @@ function Diff(hub, container, state) {
...
+                labelField: 'name',
+                searchField: ['name'],
+                options: [],
+                items: [],
+                render: {
+                    option: function (item, escape) {
+                        var origin = item.editorId !== false ? 'Editor #' + item.editorId : 'Tree #' + item.treeId;
+                        return (
+                            '<div>' +
+                            '<span class="compiler">' +
```
Comment (jeremy-rifkin) [inline static/panes/diff.js:181]: This choice from prettier is a little weird but no major concern

File: static/rison.js [hunk 2]
```diff
@@ -138,7 +144,12 @@
             },
             s = {
                 array: function (x) {
-                    var a = ['!('], b, f, i, l = x.length, v;
+                    var a = ['!('],
+                        b,
+                        f,
+                        i,
+                        l = x.length,
+                        v;
```
Comment (jeremy-rifkin) [inline static/rison.js:152]: Same here

File: static/rison.js [hunk 3]
```diff
@@ -175,9 +184,14 @@
...
-                        var a = ['('], b, f, i, v, ki, ks = [];
-                        for (i in x)
-                            ks[ks.length] = i;
+                        var a = ['('],
+                            b,
+                            f,
+                            i,
+                            v,
+                            ki,
+                            ks = [];
```
Comment (jeremy-rifkin) [inline static/rison.js:193]: Same here

**Review by jeremy-rifkin — APPROVED** [review]
Changes look good overall, I took a quick look at a small subset mainly in static/. Looks like a good improvement overall I just marked a couple things that looked weird.

File: static/rison.js (hunk 2 above)
Comment (mattgodbolt) [reply static/rison.js:152]: argh yeah good spot. I'll ignore this one (or come up with something here)

File: static/panes/diff.js (hunk 1 above)
Comment (mattgodbolt) [reply static/panes/diff.js:181]: Agreed; I think I'll leave this as-is. I've seen worse, and frankly we could phrase this a lot better now we're transpiling all the things too. But ... for another PR!

File: package.json [hunk 4]
```diff
@@ -168,10 +168,10 @@
     "sentry": "npx -p @sentry/cli sentry-cli",
     "update-browserslist": "npx browserslist@latest -- --update-db",
     "prepare": "husky install",
-    "format": "prettier --ignore-unknown ./static/**/*.ts",
-    "format-fix": "prettier --write --ignore-unknown ./static/**/*.ts",
+    "format": "prettier --ignore-unknown .",
```
Comment (mattgodbolt) [inline package.json:171]: CC @RubenRBS this is worth a second look (even post-hoc if I merge)

File: static/panes/diff.js (hunk 1 above)
Comment (jeremy-rifkin) [reply static/panes/diff.js:181]: Good point!

**Comment by mattgodbolt** [comment]
This is a biggie, so... may just merge now and we can discuss any fall out. Things I did:

* Checked all the docs have no visible differences (GH diff makes this easy) OR have improved formatting (I had to annotate some code blocks with the right language else it got formatted as that language, and we had some copy/paste blocks with the wrong language)
* Ran locally and clicked about
* Checked that the cookies and privacy policy docs hash didn't change (though a check in staging would be good)

**Review by matsjla — COMMENTED** [review]
Great stuff!!

**Pull request merged**